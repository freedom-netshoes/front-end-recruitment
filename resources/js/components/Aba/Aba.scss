.aba{
    position: absolute;
    top: 0;
    right: 0;
    width: 10px;
    height: 100%;
    background-color: #202025;     
}