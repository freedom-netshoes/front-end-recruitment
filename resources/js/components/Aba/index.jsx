import React, {Component, PropTypes} from 'react';
import CSSModules from 'react-css-modules';
import styles from './Aba.scss';

@CSSModules(styles)
class Aba extends Component {
  render () {
    const {abrirCarrinho} = this.props;

    return (
      <div styleName='aba' onMouseOver={abrirCarrinho} />
    );
  }
}

Aba.defaultProps = {
  abrirCarrinho: () => {}
};

Aba.propTypes = {
  abrirCarrinho: PropTypes.func
};

export default Aba;
