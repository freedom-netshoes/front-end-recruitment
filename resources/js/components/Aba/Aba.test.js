'use strict';

import { shallow } from 'enzyme';
import React from 'react';
import Aba from './';

it('Should Aba match to snapshot without props', () => {
  let tree = shallow(<Aba />);
  expect(tree).toMatchSnapshot();
});

it('Should Aba match to snapshot with mouseover anonymous func', () => {
  let tree = shallow(<Aba abrirCarrinho={() => {}} />);
  expect(tree).toMatchSnapshot();
});

it('Should Aba match to snapshot simulating mouseover', () => {
  let tree = shallow(<Aba abrirCarrinho={() => {}} />);
  tree.find('.aba').simulate('mouseover');
  expect(tree).toMatchSnapshot();
});