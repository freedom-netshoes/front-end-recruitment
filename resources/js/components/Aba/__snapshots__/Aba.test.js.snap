// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Aba match to snapshot simulating mouseover 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="aba"
    onMouseOver={[Function]}
/>,
  "nodes": Array [
    <div
      className="aba"
      onMouseOver={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Aba
        abrirCarrinho={[Function]}
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WrappedComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "abrirCarrinho": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="aba"
          onMouseOver={[Function]}
/>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="aba"
          onMouseOver={[Function]}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Aba
    abrirCarrinho={[Function]}
/>,
}
`;

exports[`Should Aba match to snapshot with mouseover anonymous func 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="aba"
    onMouseOver={[Function]}
/>,
  "nodes": Array [
    <div
      className="aba"
      onMouseOver={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Aba
        abrirCarrinho={[Function]}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WrappedComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "abrirCarrinho": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="aba"
          onMouseOver={[Function]}
/>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="aba"
          onMouseOver={[Function]}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Aba
    abrirCarrinho={[Function]}
/>,
}
`;

exports[`Should Aba match to snapshot without props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="aba"
    onMouseOver={[Function]}
/>,
  "nodes": Array [
    <div
      className="aba"
      onMouseOver={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Aba
        abrirCarrinho={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WrappedComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "abrirCarrinho": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="aba"
          onMouseOver={[Function]}
/>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="aba"
          onMouseOver={[Function]}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Aba
    abrirCarrinho={[Function]}
/>,
}
`;
